src/react/components/CardMarkdownContent.tsx

```diff
@@ -0,0 +1,58 @@
+import { TFile } from "obsidian";
+import { Fragment } from "preact";
+import { useState, useRef, useEffect } from "preact/hooks";
+
+import CardMarkdownContentRenderer from "~/react/components/CardMarkdownContentRenderer";
+import { useAppMount } from "~/react/context/app-mount-provider";
+import { useIntersectionObserver } from "~/react/utils/intersection-observer";
+
+interface CardMarkdownContentProps {
+  file: TFile;
+}
+
+export default function CardMarkdownContent(props: CardMarkdownContentProps) {
+  const { app } = useAppMount();
+  const { file } = props;
+  const { vault } = app;
+  const [content, setContent] = useState("");
+
+  const ref = useRef<HTMLDivElement | null>(null);
+  const entry = useIntersectionObserver(ref, {
+    rootMargin: "33%",
+    freezeOnceVisible: true,
+  });
+  const isVisible = !!entry?.isIntersecting;
+  const linesToKeep = 30;
+
+  useEffect(() => {
+    (async () => {
+      if (isVisible) {
```

Comment:
[♻️](https://github.com/erikthedeveloper/code-review-emoji-guide#:~:text=:recycle:) You can remove 1 level of nesting by doing an early return `if (!isVisible)`